src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -262,6 +262,18 @@ typedef struct DistributedExecution
 
 	/* statistics on distributed execution */
 	DistributedExecutionStats *executionStats;
+
+	/*
+	 * Used while receiving results from remote nodes. We store this information
+	 * here to avoid re-allocating it every time.
+	 *
+	 * columnArray field is reset/calculated per row, so might be useless for other
+	 * contextes. The benefit of keeping it here is to avoid allocating the array
+	 * over and over again.
+	 */
+	AttInMetadata *attributeInputMetadata;
+	uint32 expectedColumnCount;
```

Comment:
Do we need to store this separately?

Reply:
No, we can use `tupleDesc->nattr`, I'll switch  to that